### PR TITLE
Force completion labels to not have newlines

### PR DESCRIPTION
**Pull request by WeetHet**
Before:
image

After:
Screenshot 2024-12-02 at 19 06 01

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -85,7 +85,7 @@ impl zed::Extension for ScalaExtension {
             CompletionKind::Keyword => "",
             _ => return None,
         };
-        let name = completion.label;
+        let name = completion.label.replace("  ", " ").replace("\n", "");
```
Comment (tgodzik) [inline src/lib.rs:88]: I don't think labels currently should have newlines, do you have an example where it happens?

Comment (WeetHet) [reply src/lib.rs:88]: The `toList` example above is one

Comment (tgodzik) [reply src/lib.rs:88]: Could you post a snippet? I can't see to reproduce the same behaviour.

https://github.com/user-attachments/assets/744bb159-7a3c-41cd-87f3-0b052e528d8c

Comment (WeetHet) [reply src/lib.rs:88]: Try `toList` on a tuple

Comment (tgodzik) [reply src/lib.rs:88]: That works, it's a bug in Metals. We should not show that in the label.

Comment (WeetHet) [reply src/lib.rs:88]: Could we merge this for now, because it wouldn't make things worse if there are no newlines and would improve the situation for now

Comment (tgodzik) [reply src/lib.rs:88]: Aure, but I wonder about the nix related stuff. Is that intended?

Comment (WeetHet) [reply src/lib.rs:88]: I've added it as an addition. If you don't want it, I can remove it

**Pull request merged**